TST: Test inequality w/ complex dtypes (GH28050)

calvh opened the pull request:
- [x] closes #28050
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Whats new::
Add a test to `pandas/test/dtypes/test_dtypes.py`

calvh committed:
TST: Test inequality w/ complex dtypes (GH28050)

Should raise TypeError

calvh committed:
Add test for complex dtype comparison using '<'

phofl commented on pandas/tests/dtypes/test_dtypes.py:
```diff
@@ -1095,3 +1095,22 @@ def test_period_dtype_compare_to_string():
+    # GH 28050
+    # https://github.com/pandas-dev/pandas/issues/28050
+    arr = np.arange(5).astype(np.complex128)
+    df = pd.DataFrame(arr)
```
Could define this in one go. We don not need the https link

phofl replied:
Please put the arange call into the DataFrame() call, otherwise lgtm

calvh replied:
> Please put the arange call into the DataFrame() call, otherwise lgtm

Done - thanks for the lessons!

phofl commented on pandas/tests/dtypes/test_dtypes.py:
```diff
@@ -1095,3 +1095,22 @@ def test_period_dtype_compare_to_string():
+
+    with pytest.raises(
+        TypeError,
+        match="'<' not supported between instances of 'complex' and 'complex'",
```
Please define match msg in a variable to avoid duplication

calvh committed:
Refactor code to reduce repetition

Remove https link to issue (keep issue number)
Save match message in variable for multiple use

calvh commented:
@phofl thanks for the comments please see revision

calvh committed:
Combine arange call with DataFrame call in test

phofl reviewed (approved):
Lgtm

mroeschke commented:
Thanks @calvh. The CI failures were unrelated.
